config/ompi_configure_options.m4

```diff
@@ -47,7 +47,7 @@ AC_ARG_ENABLE([mpi],
 
 AC_MSG_CHECKING([if want compile-time warnings inside of mpi.h])
 AC_ARG_ENABLE(mpi-interface-warning,
```

Comment:
While you're making this change, can you fix the quoting as well?  I noticed a couple of places in this PR where we're missing quoting.

Reply:
See added commits:
* Updated the libtool macros.
* Tried to catch all missing quotings for AC_ARG_* for ompi's config files
Here my regex trying to find (hopefully) all places:
```
git grep -P 'A[CM]_ARG_[A-Z0-9_]+\([^[\)][a-z]' configure.ac config/*.m4
git grep -P '[^[]AS_HELP_STRING'
```